api/types/constants.go

```diff
@@ -573,18 +573,28 @@ const (
 	// KindStaticHostUser is a host user to be created on matching SSH nodes.
 	KindStaticHostUser = "static_host_user"
 
+	// KindIdentityCenter is an umbrella kind, representing all KindIdentityCenter*
+	// resource kinds in RBAC checks. This is to simplify Role condition statements
+	// so that they don't have to individually specify all of the Identity Center
+	// resource kinds.
+	KindIdentityCenter = "aws_identity_center"
+
 	// KindIdentityCenterAccount describes an Identity-Center managed AWS Account
+	// DO NOT USE THIS KIND IN RBAC CHECKS: use KindIdentityCenter instead
```

Comment:
This seems error prone. 

What if we made these constants the SubKind?

Reply:
I don't know if we need to make these sub-kinds (that would probably complicate existing resources/caching quite a bit) but @tcsc what we usually do in these cases is in the ACL layer you check whether identity has either specific permission or the more broad one so e.g. in the service methods that deal with IC accounts, you would see check KindIdentityCenterAccount or KindIdentityCenter. We do this with e.g. connectors - there are kinds for each specific connector type and the kind that covers all auth connectors.

I also don't see where this new kind is being used, only that it's added to the default roles - is that PR coming to teleport.e next?